addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/KM200ThingType.java

```diff
@@ -0,0 +1,356 @@
+        }
+
+        @Override
+        public List<String> ignoreSubService() {
+            List<String> subServices = new ArrayList<String>();
+            return subServices;
+        }
+
+        @Override
+        public String getActiveCheckSubPath() {
+            return null;
+        }
+
+        @Override
+        public List<String> asBridgeProperties() {
+            List<String> asProperties = new ArrayList<String>();
+            return asProperties;
+        }
+    },
+    SENSOR {
+        @Override
+        public String getRootPath() {
+            return "/system/sensors";
+        }
+
+        @Override
+        public ThingTypeUID getThingTypeUID() {
+            return KM200BindingConstants.THING_TYPE_SYSTEM_SENSOR;
+        }
+
+        @Override
+        public List<String> ignoreSubService() {
+            List<String> subServices = new ArrayList<String>();
+            return subServices;
+        }
+
+        @Override
+        public String getActiveCheckSubPath() {
+            return null;
+        }
+
+        @Override
+        public List<String> asBridgeProperties() {
+            List<String> asProperties = new ArrayList<String>();
+            return asProperties;
+        }
+    },
+    SYSTEM {
+        @Override
+        public String getRootPath() {
+            return "/system";
+        }
+
+        @Override
+        public ThingTypeUID getThingTypeUID() {
+            return KM200BindingConstants.THING_TYPE_SYSTEM;
+        }
+
+        @Override
+        public List<String> ignoreSubService() {
+            List<String> subServices = new ArrayList<String>();
+            subServices.add("sensors");
+            subServices.add("appliance");
+            subServices.add("holidayModes");
+            return subServices;
+        }
+
+        @Override
+        public String getActiveCheckSubPath() {
+            return null;
+        }
+
+        @Override
+        public List<String> asBridgeProperties() {
+            List<String> asProperties = new ArrayList<String>();
+            asProperties.add("bus");
+            asProperties.add("systemType");
+            asProperties.add("brand");
+            asProperties.add("info");
+            return asProperties;
+        }
+    },
+    SWITCHPROGRAM {
+        @Override
+        public String getRootPath() {
+            return "";
+        }
+
+        @Override
+        public ThingTypeUID getThingTypeUID() {
+            return KM200BindingConstants.THING_TYPE_SWITCH_PROGRAM;
+        }
+
+        @Override
+        public List<String> ignoreSubService() {
+            List<String> subServices = new ArrayList<String>();
+            return subServices;
+        }
+
+        @Override
+        public String getActiveCheckSubPath() {
+            return null;
+        }
+
+        @Override
+        public List<String> asBridgeProperties() {
+            List<String> asProperties = new ArrayList<String>();
```

Comment:
Maybe its becomes way better if you only override what is different, so returning and empty list seems the default, so implement the method in the main enum to return `Collections.emptyList();` (instead of that empty array list.) I can image that by removing the defaults implementations we can can already get closer to a nice solution.

Reply:
I would then let the `getRootPath` & `getThingTypeUID` work via the constructor as I suggested above.

Reply:
I changed the enums and the handling. All data are now in one enum and it is extended with the contructor and default data handling. Please take a look....

Reply:
Yes, this is better, thanks!